mysql/datadog_checks/mysql/databases_data.py

```diff
@@ -0,0 +1,370 @@
+        list of table dicts
+        "name": str
+        """
+        self._cursor_run(cursor, query=SQL_TABLES, params=db_name)
+        tables_info = fetch_and_convert_to_str(cursor)
+        return tables_info
+
+    @tracked_method(agent_check_getter=agent_check_getter, track_result_length=True)
+    def _get_tables_data(self, table_list, db_name, cursor):
+
+        if len(table_list) == 0:
+            return
+        table_name_to_table_index = {}
+        table_names = ""
+        for i, table in enumerate(table_list):
+            table_name_to_table_index[table["name"]] = i
+            table_names += '"' + str(table["name"]) + '",'
+        table_names = table_names[:-1]
+        total_columns_number = self._populate_with_columns_data(
+            table_name_to_table_index, table_list, table_names, db_name, cursor
+        )
+        self._populate_with_partitions_data(table_name_to_table_index, table_list, table_names, db_name, cursor)
+        self._populate_with_foreign_keys_data(table_name_to_table_index, table_list, table_names, db_name, cursor)
+        self._populate_with_index_data(table_name_to_table_index, table_list, table_names, db_name, cursor)
+        return total_columns_number, table_list
+
+    @tracked_method(agent_check_getter=agent_check_getter)
+    def _populate_with_columns_data(self, table_name_to_table_index, table_list, table_names, db_name, cursor):
+        self._cursor_run(
+            cursor,
+            query=SQL_COLUMNS.format(table_names),
+            params=db_name,
+        )
+        rows = fetch_and_convert_to_str(cursor)
+        for row in rows:
+            if "nullable" in row:
+                if row["nullable"].lower() == "yes":
+                    row["nullable"] = True
+                else:
+                    row["nullable"] = False
+            table_name = str(row.pop("table_name"))
+            table_list[table_name_to_table_index[table_name]].setdefault("columns", [])
+            table_list[table_name_to_table_index[table_name]]["columns"].append(row)
+
+        return len(rows)
+
+    @tracked_method(agent_check_getter=agent_check_getter)
+    def _populate_with_index_data(self, table_name_to_table_index, table_list, table_names, db_name, cursor):
+        self._cursor_run(cursor, query=SQL_INDEXES.format(table_names), params=db_name)
+        rows = fetch_and_convert_to_str(cursor)
+        for row in rows:
+            table_name = str(row.pop("table_name"))
+            table_list[table_name_to_table_index[table_name]].setdefault("indexes", [])
+            if "nullables" in row:
+                nullables_arr = row["nullables"].split(',')
+                nullables_converted = ""
+                for s in nullables_arr:
+                    if s.lower() == "yes":
+                        nullables_converted += "true,"
+                    else:
+                        nullables_converted += "false,"
+                row["nullables"] = nullables_converted[:-1]
```

Comment:
I'm slightly concerned about this statement, which uses the last value from the nullables_arr to indicate if the entire index is nullable or not. Assume a mixed `yes/no` values in nullables column happens when it's a compound index. using the last nullable value from the array seems error prone. Is it better to retain the original value of the nullables_arr so that it's clear which column is actually nullable in a compound index?

Reply:
By `nullables_converted[:-1]` I just removed the last coma, not the last value cause `nullables_converted` is a string. 
I hesitated to check if we are at the last element of the `nullables_arr` in order to add the "," but decided just to remove the extra coma at the end of the loop.

Reply:
what's the benefit of converting `yes/no` string to `true/false`?  If we truly want to do the conversion, how about making column `nullables` an array of booleans?

Reply:
closing as discussed